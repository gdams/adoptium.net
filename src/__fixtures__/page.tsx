import {
    AllReleaseNotes,
    AsciidocPage,
    File,
    ReleaseNotePage
} from '../types';

export const createAsciidocData = (): {
    asciidoc: AsciidocPage;
    file: File;
  } => ({
    asciidoc: {
        id: 'asciidoc-1',
        html: `
            <p>Hello World, this is a <pre><code>test</code></pre></p>
            <table><thead><td>Header 1</td><td>Header 2</td></thead><tbody><tr><td>Cell 1</td><td class="icon">Cell 2</td></tr></tbody></table>
            <a href="https://www.eclipse.org">Eclipse</a>
            <span class="icon">[docker]</span>
            <span class="icon">[check]</span>
        `,
        document: {
            title: 'Asciidoc Page title',
            main: 'Asciidoc Page content',
        },
        pageAttributes: {
            authors: 'author1, author2',
        },
        fields: {
            slug: '/asciidoc/asciidoc-page-title',
        }
    },
    file: {
        relativePath: 'test.adoc',
    }
});

export const createReleaseNoteData = (): {
    allReleaseNotes: AllReleaseNotes;
  } => ({
    allReleaseNotes: {
        edges: [
            {
                node: {
                    id: 'release-note-1',
                    version: '1.0.0',
                    title: 'Release Note title',
                    priority: 1,
                    component: 'component1',
                    link: 'https://release-note.com',
                }
            }
        ]
    }
});

