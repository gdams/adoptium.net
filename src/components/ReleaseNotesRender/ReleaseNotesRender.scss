.MuiTablePagination-toolbar p {
  margin: 0 !important;
}

.MuiDataGrid-root .MuiDataGrid-cell {
  white-space: normal !important;
  word-wrap: break-word !important;
  text-align: left !important;
}

.priority-1 {
  background-color: #e5121a !important;
}

.priority-2 {
  background-color: #ec8448 !important;
}

.priority-3 {
  background-color: #27ea51 !important;
}

.priority-4 {
  background-color: #0088e7 !important;
}

.priority-5 {
  background-color: #a400b0 !important;
}
