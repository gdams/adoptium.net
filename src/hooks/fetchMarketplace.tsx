const baseUrl = 'https://staging-marketplace-api.adoptium.net';
    
export async function getAllPkgsForVersion(
        version: number,
        os: string,
        architecture: string,
        package_type: string,
        checkboxRef: any,
    ): Promise<apiData> {
    let microsoftSelected = checkboxRef.current.vendorMicrosoft.checked;
    let temurinSelected = checkboxRef.current.vendorAdoptium.checked;
    let redhatSelected = checkboxRef.current.vendorRedHat.checked;
    let huaweiSelected = checkboxRef.current.vendorHuawei.checked;
    let zuluSelected = checkboxRef.current.vendorAzul.checked;
    let ibmSelected = checkboxRef.current.vendorIBM.checked;

    let params = '?'
    params += 'feature_version=' + version;

    if (os !== 'any' ) {
        if (os === 'alpine-linux') {
            params += '&os=alpine_linux'
        } else {
            params += ('&os=' + os)
        }
    }

    if (architecture !== 'any' ) {
        params += ('&architecture=' + architecture)
    }

    if (package_type === 'any') {
        params += '&image_type=jdk&image_type=jre'
    } else {
        params += ('&image_type=' + package_type)
    }

    if (temurinSelected) {
        params += '&vendor=adoptium'
    }

    if (redhatSelected) {
        params += '&vendor=redhat'
    }

    if (huaweiSelected) {
        params += '&vendor=huawei'
    }

    if (microsoftSelected) {
        params += ('&vendor=microsoft')
    }

    if (zuluSelected) {
        params += ('&vendor=azul')
    }

    if (ibmSelected) {
        params += ('&vendor=ibm')
    }

    let url = baseUrl + '/v1/assets/latestForVendors' + params;
    let json = await getPkgs(url);
    const data = JSON.parse(json);
    return data
}

async function getPkgs(url) {
    let response = await makeRequest("GET", url);
    return response;
}

async function makeRequest(method, url): Promise<apiData> {
    const response = await fetch(url);
    const apiResult = await response.text();
    return apiResult
};

export function getImageForDistribution(distribution: string) {
    switch(distribution) {
    case 'microsoft': return '/images/microsoft-logo.png';
    case 'temurin': return '/images/adoptium-logo.png';
    case 'redhat': return '/images/redhat.svg';
    case 'bisheng': return '/images/huawei.svg';
    case 'zulu': return '/images/azul-logo.png';
    case 'semeru_certified': return '/images/ibm-logo.png';
    default: return '';
    }
}
