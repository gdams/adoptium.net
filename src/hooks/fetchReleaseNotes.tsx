import { VersionMetaData } from '.';
import { useEffect, useState } from 'react';

const baseUrl = 'https://api.adoptium.net/v3/info/release_notes';

export function fetchReleaseNotesForVersion(
    isVisible: boolean,
    version: any,
): ReleaseNoteAPIResponse | null {
    if (!version) {
        return null
    }

    const [releaseNotes, setReleaseNotes] = useState<ReleaseNoteAPIResponse | null>(null);
    useEffect(() => {
        if (isVisible) {
        (async () => {
            setReleaseNotes(await fetchReleaseNote(version));
        })();
        }
    }, [isVisible]);

    return releaseNotes;
}

async function fetchReleaseNote(version) {
    const url = `${baseUrl}/${version}`;
    try {
        // fetch the data from the API
        const response = await fetch(url);
        return await response.json();
    } catch (error) {
        return { release_notes: null };
    }
}

export interface ReleaseNoteAPIResponse {
    id: string;
    release_name: string;
    release_notes: ReleaseNote[];
    vendor: string;
    version_data: VersionMetaData;
}

export interface ReleaseNote {
    id: string;
    link: URL;
    title: string;
    backportOf?: string;
    priority?: string;
    component?: string;
    subcomponent?: string;
    type?: string;
}
