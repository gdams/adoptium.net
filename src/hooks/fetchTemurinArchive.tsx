import moment from "moment";

import { fetchExtension } from '../util/fetchExtension';

const baseUrl = 'https://api.adoptium.net/v3';

let releases = []

export async function getAssetsForVersion(version: any, releaseType: any, numBuilds: any, buildDate: Date) {
  let url = `${baseUrl}/assets/feature_releases/${version}/${releaseType}?vendor=eclipse`;
  if (numBuilds) {
    url += `&page_size=${numBuilds}`
  }
  if (buildDate) {
    url += `&before=${moment(buildDate).format('Y-MM-DD')}`
  }
  releases = []
  let   json      = await makeRequest('GET', url);
  const response  = JSON.parse(json);
  const data      = response;
  let   pkgsFound = []
  for (let i = 0 ; i < data.length ; i++) {
      pkgsFound.push(data[i]);
  }
  return renderReleases(pkgsFound);
}

async function makeRequest(method: string, url: RequestInfo): Promise<apiData> {
    const response = await fetch(url);
    const apiResult = await response.text();
    return apiResult
};

function renderReleases(pkgs) {
  pkgs.forEach((aRelease) => {
    const release = {
        release_name: aRelease.release_name,
        openjdk_version: aRelease.version_data.openjdk_version,
        release_link: aRelease.release_link,
        timestamp: aRelease.timestamp,
        platforms: {},
    };

    aRelease.binaries.forEach((aReleaseAsset: { os: any; architecture: any; image_type: string; package: { link: any; checksum: any; size: number; }; installer: { link: any; checksum: any; size: number; }; }) => {
        const platform = `${aReleaseAsset.os}-${aReleaseAsset.architecture}`
  
        // Skip this asset if its platform could not be matched (see the website's 'config.json')
        if (!platform) {
          return;
        }
  
        // Skip this asset if it's not a binary type we're interested in displaying
        const binary_type = aReleaseAsset.image_type.toUpperCase();
        if (binary_type == 'SOURCES') {
          release.source_url = aReleaseAsset.package.link;
        }
        if (!['INSTALLER', 'JDK', 'JRE'].includes(binary_type)) {
          return;
        }
  
        if (!release.platforms[platform]) {
          release.platforms[platform] = {
            assets: [],
          };
        }
  
        let binary_constructor = {
          os: aReleaseAsset.os,
          architecture: aReleaseAsset.architecture,
          type: binary_type,
          link: aReleaseAsset.package.link,
          checksum: aReleaseAsset.package.checksum,
          size: Math.floor(aReleaseAsset.package.size / 1000 / 1000),
          extension: fetchExtension(aReleaseAsset.package.link)
        };
  
        if (aReleaseAsset.installer) {
          binary_constructor.installer_link = aReleaseAsset.installer.link;
          binary_constructor.installer_checksum = aReleaseAsset.installer.checksum;
          binary_constructor.installer_size =  Math.floor(aReleaseAsset.installer.size / 1000 / 1000);
          binary_constructor.installer_extension = fetchExtension(aReleaseAsset.installer.link)
        }
  
        // Add the new binary to the release asset
        release.platforms[platform].assets.push(binary_constructor);
    });
    releases.push(release);
  })
  return releases
}
