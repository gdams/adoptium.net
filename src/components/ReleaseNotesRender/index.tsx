import React, { useRef, MutableRefObject } from 'react';
import { DataGrid, GridColDef, GridToolbarContainer, GridToolbarFilterButton, gridClasses } from '@mui/x-data-grid';
import { useLocation } from '@reach/router';
import queryString from 'query-string';

import { fetchReleaseNotesForVersion, useOnScreen } from '../../hooks';
import './ReleaseNotesRender.scss';

export const fetchTitle = (priority) => {
  let title
  switch (priority) {
    case '1':
      title = 'P1 - Blocks development and/or testing work, production could not run.';
      break;
    case '2':
      title = 'P2 - Crashes, loss of data, severe memory leak.';
      break;
    case '3':
      title = 'P3 - Major loss of function.';
      break;
    case '4':
      title = 'P4 - Minor loss of function, or other problem where easy workaround is present.';
      break;
    case '5':
      title = 'P5 - Cosmetic problem like misspelt words or misaligned text.';
      break;
    case '6':
      title = 'This issue is not publicly visible.'
  }
  return title;
};

const CustomToolbar: React.FunctionComponent<{
  setFilterButtonEl: React.Dispatch<React.SetStateAction<HTMLButtonElement | null>>;
}> = ({ setFilterButtonEl }) => (
  <GridToolbarContainer>
    <GridToolbarFilterButton ref={setFilterButtonEl} />
  </GridToolbarContainer>
);

const columns: GridColDef[] = [
  {
    field: 'priority',
    type: 'singleSelect',
    valueOptions: ['1', '2', '3', '4', '5'],
    headerName: 'Priority',
    width: 100,
    renderCell: (params) => {
      const title = fetchTitle(params.value);
      return (
        // check if params.value is defined, if not, return an empty string
        <span title={title} className={`badge bg-secondary priority-${params.value}`}>{params.value && params.value !== '6' ? `P${params.value}` : 'P?'}</span>
      )
    },
  },
  {
    field: 'type',
    type: 'singleSelect',
    // TODO: This needs automatically setting
    valueOptions: ['Backport', 'Bug', 'Enhancement'],
    headerName: 'Type',
    width: 100,
    sortable: false,
    renderCell: (params) => (
      <span>{params.value ? params.value : 'Hidden'}</span>
    )
  },
  {
    field: 'component',
    headerName: 'Component',
    width: 150,
    sortable: false
  },
  {
    field: 'id',
    headerName: 'Issue',
    width: 150,
    sortable: false,
    renderCell: (params) => (
      <a target='_blank' rel='noopener noreferrer' href={params.row.link}>{params.value}</a>
    ),
  },
  { field: 'title', headerName: 'Title', width: 800, sortable: false },
];

const ReleaseNotesRender = (): null | JSX.Element => {
  const version = queryString.parse(useLocation().search, {decode: false}).version;

  const ref = useRef<HTMLDivElement | null>(null);
  const isVisible = useOnScreen(ref as MutableRefObject<Element>, true);
  const releaseNotes = fetchReleaseNotesForVersion(isVisible, version);

  // Set all priorities set as undefined to '?' to avoid errors
  releaseNotes?.release_notes?.forEach((note) => {
    if (note.priority === undefined) {
      note.priority = '6';
    }
  });

  interface FilterItem {
    field: string;
    operator: string;
    value: string;
  }

  // Set type to 'Enhancement' by default if version matches jdk-xx+xx
  const regex = /^jdk-(2\d|\d{3,})\+\d+$/;
  let filterItems: FilterItem[] = [];
  if (version?.toString && regex.test(version?.toString())) {
    filterItems.push(
      {
        field: 'type',
        operator: 'is',
        value: 'Enhancement'
      }
    )
  }

  const totalP1 = releaseNotes?.release_notes?.filter((note) => note.priority === '1').length;

  return (
	  <div ref={ref} className='text-center'>
    <h2>{version}</h2>
      <div className='pt-3' style={{ display: 'flex', height: '100%' }}>
        <div style={{ flexGrow: 1 }}>
          {releaseNotes?.release_notes === null ? (
            <>
            <h2>Oops... We couldn't find those release notes</h2>
            <span>Please ensure that you have a specified a version using the version URL parameter: <code>?version=x.x.x</code></span>
            </>
          ) : (
            <>
              <p>This section organizes the changes in the selected update release by the main component under which each issue is filed.</p>
              <p><strong>{`The total number of fixes marked as P1 is: ${totalP1}`}</strong></p>
              <DataGrid
                aria-label='Release Notes'
                autoHeight
                rows={releaseNotes && releaseNotes.release_notes? releaseNotes.release_notes : []}
                loading={releaseNotes === null}
                columns={columns}
                initialState={{
                  pagination: {
                    paginationModel: {
                      pageSize: 20
                    }
                  },
                  filter: {
                    filterModel: {
                      items: filterItems
                    }
                  },
                  sorting: {
                    sortModel: [{ field: 'priority', sort: 'asc' }],
                  },
                }}
                pageSizeOptions={[20, 50, 75]}
                pagination
                isRowSelectable={() => false}
                components={{
                  Toolbar: CustomToolbar,
                }}
                getRowHeight={() => 'auto'}
                sx={{
                  [`& .${gridClasses.cell}`]: {
                    py: 1,
                  },
                }}
              />
            </>
          )}
        </div>
      </div>
    </div>
  );
 };

export default ReleaseNotesRender;
