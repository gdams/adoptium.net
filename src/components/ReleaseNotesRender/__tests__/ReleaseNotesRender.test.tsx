import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest'
import ReleaseNotesRender, { fetchTitle } from '../index';
import { fetchReleaseNotesForVersion } from '../../../hooks/fetchReleaseNotes';
import { createMockReleaseNotesAPI  } from '../../../__fixtures__/hooks';

vi.mock('../../../hooks/fetchReleaseNotes');

afterEach(() => {
    vi.clearAllMocks();
});

describe('ReleaseNotesRender component', () => {
    it('should render correctly - version not defined', () => {
        const { container } = render(
            <ReleaseNotesRender />
        );
        expect(container).toMatchSnapshot();
    });

    it('fetchTitle should return correct title', () => {
        expect(fetchTitle('1')).toContain('P1');
        expect(fetchTitle('2')).toContain('P2');
        expect(fetchTitle('3')).toContain('P3');
        expect(fetchTitle('4')).toContain('P4');
        expect(fetchTitle('5')).toContain('P5');
        expect(fetchTitle('6')).toContain('not publicly visible');
        expect(fetchTitle(null)).toBeUndefined();
        expect(fetchTitle('123')).toBeUndefined();
    });
    
    it('should render correctly', () => {
        // mock query string version
        vi.mock('query-string', () => ({
            default: {
              parse: () => ({
                version: 'version',
              }),
            }
        }));
        fetchReleaseNotesForVersion.mockReturnValue(createMockReleaseNotesAPI(1));
        const { container } = render(
            <ReleaseNotesRender />
        );
        expect(fetchReleaseNotesForVersion).toHaveBeenCalledTimes(1);
        expect(container).toMatchSnapshot();
    });

    it('should render correctly - no release notes', () => {
        // mock query string version
        vi.mock('query-string', () => ({
            default: {
              parse: () => ({
                version: 'version',
              }),
            }
        }));
        fetchReleaseNotesForVersion.mockReturnValue({ release_notes: null});
        const { container } = render(
            <ReleaseNotesRender />
        );
        expect(fetchReleaseNotesForVersion).toHaveBeenCalledTimes(1);
        expect(container).toMatchSnapshot();
    });
});