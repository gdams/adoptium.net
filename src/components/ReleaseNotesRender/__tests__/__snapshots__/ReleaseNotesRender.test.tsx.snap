// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReleaseNotesRender component > should render correctly - no release notes 1`] = `
<div>
  <div
    class="text-center"
  >
    <h2>
      version
    </h2>
    <div
      class="pt-3"
      style="display: flex; height: 100%;"
    >
      <div
        style="flex-grow: 1;"
      >
        <h2>
          Oops... We couldn't find those release notes
        </h2>
        <span>
          Please ensure that you have a specified a version using the version URL parameter: 
          <code>
            ?version=x.x.x
          </code>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`ReleaseNotesRender component > should render correctly - version not defined 1`] = `
<div>
  <div
    class="text-center"
  >
    <h2>
      version
    </h2>
    <div
      class="pt-3"
      style="display: flex; height: 100%;"
    >
      <div
        style="flex-grow: 1;"
      >
        <p>
          This section organizes the changes in the selected update release by the main component under which each issue is filed.
        </p>
        <p>
          <strong>
            The total number of fixes marked as P1 is: undefined
          </strong>
        </p>
        <div
          aria-colcount="5"
          aria-label="Release Notes"
          aria-multiselectable="false"
          aria-rowcount="1"
          class="MuiDataGrid-root MuiDataGrid-autoHeight MuiDataGrid-root--densityStandard MuiDataGrid-withBorderColor css-1bvp8j9-MuiDataGrid-root"
          role="grid"
        >
          <div
            class="MuiDataGrid-toolbarContainer css-128fb87-MuiDataGrid-toolbarContainer"
          >
            <button
              aria-label="Show filters"
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
              >
                <span
                  class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FilterListIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
                    />
                  </svg>
                  <span
                    class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary css-1pi4uwz-MuiBadge-badge"
                  >
                    0
                  </span>
                </span>
              </span>
              Filters
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
          >
            <div
              class="MuiDataGrid-columnHeaders MuiDataGrid-withBorderColor css-1iyq7zh-MuiDataGrid-columnHeaders"
              style="min-height: 56px; max-height: 56px; line-height: 56px;"
            >
              <div
                class="MuiDataGrid-columnHeadersInner css-gl260s-MuiDataGrid-columnHeadersInner"
                role="rowgroup"
              />
            </div>
          </div>
          <div
            class="MuiDataGrid-footerContainer MuiDataGrid-withBorderColor css-wop1k0-MuiDataGrid-footerContainer"
          >
            <div />
            <div
              class="MuiTablePagination-root css-rtrcn9-MuiTablePagination-root"
            >
              <div
                class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
              >
                <div
                  class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
                />
                <p
                  class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                  id=":r2:"
                >
                  Rows per page:
                </p>
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby=":r2: :r1:"
                    class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                    id=":r1:"
                    role="button"
                    tabindex="0"
                  >
                    20
                  </div>
                  <input
                    aria-hidden="true"
                    aria-invalid="false"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    tabindex="-1"
                    value="20"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
                <p
                  class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
                >
                  0â€“0 of 0
                </p>
                <div
                  class="MuiTablePagination-actions"
                >
                  <button
                    aria-label="Go to previous page"
                    class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                    disabled=""
                    tabindex="-1"
                    title="Go to previous page"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowLeftIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                      />
                    </svg>
                  </button>
                  <button
                    aria-label="Go to next page"
                    class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                    disabled=""
                    tabindex="-1"
                    title="Go to next page"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowRightIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ReleaseNotesRender component > should render correctly 1`] = `
<div>
  <div
    class="text-center"
  >
    <h2>
      version
    </h2>
    <div
      class="pt-3"
      style="display: flex; height: 100%;"
    >
      <div
        style="flex-grow: 1;"
      >
        <p>
          This section organizes the changes in the selected update release by the main component under which each issue is filed.
        </p>
        <p>
          <strong>
            The total number of fixes marked as P1 is: 1
          </strong>
        </p>
        <div
          aria-colcount="5"
          aria-label="Release Notes"
          aria-multiselectable="false"
          aria-rowcount="2"
          class="MuiDataGrid-root MuiDataGrid-autoHeight MuiDataGrid-root--densityStandard MuiDataGrid-withBorderColor css-1bvp8j9-MuiDataGrid-root"
          role="grid"
        >
          <div
            class="MuiDataGrid-toolbarContainer css-128fb87-MuiDataGrid-toolbarContainer"
          >
            <button
              aria-label="Show filters"
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
              >
                <span
                  class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FilterListIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
                    />
                  </svg>
                  <span
                    class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary css-1pi4uwz-MuiBadge-badge"
                  >
                    0
                  </span>
                </span>
              </span>
              Filters
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
          >
            <div
              class="MuiDataGrid-columnHeaders MuiDataGrid-withBorderColor css-1iyq7zh-MuiDataGrid-columnHeaders"
              style="min-height: 56px; max-height: 56px; line-height: 56px;"
            >
              <div
                class="MuiDataGrid-columnHeadersInner css-gl260s-MuiDataGrid-columnHeadersInner"
                role="rowgroup"
              />
            </div>
          </div>
          <div
            class="MuiDataGrid-footerContainer MuiDataGrid-withBorderColor css-wop1k0-MuiDataGrid-footerContainer"
          >
            <div />
            <div
              class="MuiTablePagination-root css-rtrcn9-MuiTablePagination-root"
            >
              <div
                class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
              >
                <div
                  class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
                />
                <p
                  class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                  id=":r5:"
                >
                  Rows per page:
                </p>
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby=":r5: :r4:"
                    class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                    id=":r4:"
                    role="button"
                    tabindex="0"
                  >
                    20
                  </div>
                  <input
                    aria-hidden="true"
                    aria-invalid="false"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    tabindex="-1"
                    value="20"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
                <p
                  class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
                >
                  1â€“1 of 1
                </p>
                <div
                  class="MuiTablePagination-actions"
                >
                  <button
                    aria-label="Go to previous page"
                    class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                    disabled=""
                    tabindex="-1"
                    title="Go to previous page"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowLeftIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                      />
                    </svg>
                  </button>
                  <button
                    aria-label="Go to next page"
                    class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                    disabled=""
                    tabindex="-1"
                    title="Go to next page"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowRightIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
